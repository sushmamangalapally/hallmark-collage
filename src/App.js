import React, { Component, useEffect, useState, Fragment }  from "react";
import logo from "./logo.svg";
import Collage from "./Collage";
import Search from "./Search";
import "./stylesheet/App.css";

function App() {

  return (
        <div className="App">
          <nav>
            <div className="nav-wrapper indigo lighten-3">
              <a href="#" className="brand-logo header">
                Create Your Photo Collage
              </a>
            </div>
          </nav>
          <header class="navbar">
  <section class="navbar-section">
    <a href="..." class="navbar-brand mr-2">Spectre.css</a>
    <a href="..." class="btn btn-link">Docs</a>
    <a href="..." class="btn btn-link">GitHub</a>
  </section>
  <section class="navbar-section">
    <div class="input-group input-inline">
      <input class="form-input" type="text" placeholder="search"/>
      <button class="btn btn-primary input-group-btn">Search</button>
    </div>
  </section>
</header>

          <Search/>
      </div>
  );
}

export default App;
