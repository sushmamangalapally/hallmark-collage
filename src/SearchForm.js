import React, { Component, useState } from "react";

const SearchTerm = ({searchTerm, onSearchInput}) => (
    <input
        type="text"
        onChange={(e) => onSearchInput(e.target.value)}
        placeholder="Search Pictures"
        className="form-input"
        id="adding_photos_text"
        value={searchTerm}
    />
);


export default SearchTerm;
