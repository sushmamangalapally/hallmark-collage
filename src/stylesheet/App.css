body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale; 
  background: #e8eaf6;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace; }

*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit; 
}

html {
  box-sizing: border-box;
  /* font-size: 62.5%;  */
}

figure {
  border: 3px solid black;
  background-repeat: no-repeat;
  background-size: cover;
  /* background-color: deepPink; */
  padding: 20px;
  color: #fff;
  border-style: dotted;
}

body {
  font-family: "Nunito", sans-serif;
  color: #1b011b;
  font-weight: 300;
  line-height: 1.6;
  background: rgb(209 213 229);
}

.main-app {
  display: flex;
  height: 800px;
  padding-top: 20px;
}

.search-photos{
  width: 40%;
  margin-left: 20px;
  margin-right: 20px;
}

.background {
  color: #504e4e;
  font-weight: 500;
}

.photos-results {
  height: 80%;
  overflow: scroll;
}
.container {
  width: 60%;
  margin: 2rem auto; 
  margin-right: 10px;
}

.gallery {
  /* grid-gap: 0;
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: repeat(12, 4vw); */
      /* grid-gap: 0; */
      display: grid;
      /* grid-gap: 10px 15px; */
      grid-template-columns: repeat(12, 1fr);
      grid-template-rows: repeat(12, 4vw);
      /* display: grid; */
      /* grid-template-columns: repeat(6, 1fr); */
      /* grid-template-rows: repeat(6, 1fr); */
      /* grid-column-gap: 5px; */
      /* grid-row-gap: 5px; */
      /* display: grid; */
      /* grid-template-columns: repeat(9, 1fr); */
      /* grid-auto-rows: minmax(100px, auto); */
      grid-template-areas: "g1 g1 g2 g2 g3 g3" "g1 g1 g2 g2 g3 g3" "sd sd sd ft ft ft ft ft ft";
          /* grid-gap: 0; */
    display: grid;
    /* grid-gap: 10px 15px; */
    grid-template-columns: repeat(12, 1fr);
    grid-template-rows: repeat(12, 4vw);
    /* display: grid; */
    /* grid-template-columns: repeat(6, 1fr); */
    /* grid-template-rows: repeat(6, 1fr); */
    grid-column-gap: 9px;
    grid-row-gap: 9px;
    /* display: grid; */
    /* grid-template-columns: repeat(9, 1fr); */
    /* grid-auto-rows: minmax(100px, auto); */
    grid-template-areas: "g1 g1 g2 g2 g3 g3" "g1 g1 g2 g2 g3 g3" "sd sd sd ft ft ft ft ft ft";
}

.gallery__img {
  width: 100%;
  height: 100%;
  object-fit: cover;
  display: block; 
}

.gallery__item--1 {
  /* grid-column-start: 1;
  grid-column-end: 6;
  grid-row-start: 1;
  grid-row-end: 5; */

  grid-column-start: 1;
  grid-column-end: 5;
  grid-row-start: 1;
  grid-row-end: 5;
}

.gallery__item--2 {
  /* grid-column-start: 5;
  grid-column-end: 10;
  grid-row-start: 1;
  grid-row-end: 5; */
  grid-column-start: 5;
  grid-column-end: 9;
  grid-row-start: 1;
  grid-row-end: 5;
}

.gallery__item--3 {
  /* grid-column-start: 9;
  grid-column-end: 14;
  grid-row-start: 1;
  grid-row-end: 8; */

  grid-column-start: 9;
  grid-column-end: 14;
  grid-row-start: 1;
  grid-row-end: 7;
}

.gallery__item--4 {
  /* grid-row-start: 5;
  grid-row-end: 9;
  grid-column-start: 1;
  grid-column-end: 6; */
  grid-row-start: 5;
  grid-row-end: 9;
  grid-column-start: 1;
  grid-column-end: 5;
}

.gallery__item--5 {
  /* grid-row-start: 5;
  grid-row-end: 9;
  grid-column-start: 5;
  grid-column-end: 10; */
  grid-row-start: 5;
  grid-row-end: 9;
  grid-column-start: 5;
  grid-column-end: 9;
}

.gallery__item--6 {
  /* grid-row-start: 9;
  grid-row-end: 13;
  grid-column-start: 1;
  grid-column-end: 6; */
  grid-row-start: 9;
  grid-row-end: 15;
  grid-column-start: 1;
  grid-column-end: 5;
}

.gallery__item--7 {
  /* grid-row-start: 9;
  grid-row-end: 13;
  grid-column-start: 5;
  grid-column-end: 10; */
  grid-row-start: 9;
  grid-row-end: 15;
  grid-column-start: 5;
  grid-column-end: 9;
}

img {
  margin: 5px;
  cursor: grab;
}

a {
  margin-right: 20px;
}

.collageHeader > h1 {
  font-weight: 700;
}

.dropdown-content li>a, .dropdown-content li>span, a.collage-btn {
  color:#474950;
}

a.collage-btn {
  margin-top: 10px;
  margin-right: 0;
}

#saveCollage {
  padding: 10px 10px 20px 10px;
}

div.background-color {
    margin-top: 10px;
    margin-top: 1rem;
    margin-bottom: 1rem;
    height: 54px;
}

.gallery__item--8 {
  /* grid-row-start: 8;
  grid-row-end: 13;
  grid-column-start: 9;
  grid-column-end: 13; */
  grid-row-start: 7;
  grid-row-end: 15;
  grid-column-start: 9;
  grid-column-end: 14;
}

.center-align {
  text-align: center;
  font-size: x-large;
}

.top-align, .bottom-align {
  text-align: center;
}

.text-center {
  border: none;
}

.row {
  /* margin: 25px 0; */
}

.instructions-drag-n-drop{
  text-align: center;
}

.header {
  font-weight: 500;
  font-family: 'Abel';
}

.adjustment {
  color: #797878;
  /* font-size: small; */
  font-weight: 400;
  background: #d6d6d66e;
  border-radius: 10px;
  padding: 10px;
}

.search-photos{
background: #3e4776;
border-radius: 7px;
padding: 15px;
color: white;
font-size: 0.8vw;
/* height: 100vh; */
background: #e8eaf6;
border-radius: 7px;
padding: 15px;
color: #1b011b;
font-size: 0.8vw;
height: 100vh;
border: 1px solid #ababba;
box-shadow: 5px 5px #bbbdbe61;
display: flex;
flex-direction: column;
/* height: 1500px; */
}

.navbar {
  font-size: 25px;
  padding: 15px;
  background: white;
}

div.list-icon {
  color: black;
  cursor: pointer;
}

a.btn {
  font-size: 25px;
  /* height: 2em; */
  padding: 1vh 1vh;
  /* margin: 10px; */
  display: inline;
  border: .05rem solid #a3a1ba;
}

div.photos-results {
  background: mintcream;
  margin: 5px 0px;
  border-radius: 5px;
  background: #d1d3dd;
  border-radius: 7px;
  padding: 15px;
  color: #1b011b;
  font-size: 1.5em;
  /* height: 100vh; */
  border: 1px solid #ababba;
  box-shadow: 5px 5px #bbbdbe61;
}

div.photos-results > h2 {
  font-size: 1.5em;
}

div.search-books-input-wrapper {
  margin: 15px 0px;
  /* font-size: 1.5em; */
  /* height: 2em; */
  padding: 1vh 1vh;
}

div.container {
  margin-top: 0px;
}

label.form-label {
  font-size: 1.5em;
  display: inline;
}

input#greeting_text, div.search-books-input-wrapper input {
  font-size: 1.5em;
  /* height: 2em; */
  padding: 1vh 1vh;
  /* margin: 10px; */
  display: inline;
  height: 100%;
  margin: 10px 0pc;
}

.menu-item a {
  color: black;
}

li.menu-item {
  font-size: 20px;
  border-bottom: 1px solid lightgray;
}

.navbar {
  padding: 15px;
  background: #b2b2d9;
  font-weight: 800;
  color: mintcream;
  /* font-size: 2em; */
}

div.container button.btn {
  font-size: 1em;
  /* height: 2em; */
  padding: 1vh 1vh;
  /* margin: 10px; */
  display: inline;
  height: 6vh;
  margin: 10px 0pc;

  border-radius: 10px;
    padding-right: 5px;
    padding-left: 5px;
}

p.messageMobile {
  display: none;
}

/* For mobile phones: */
@media only screen and (max-device-width: 480px) {
  div.search-photos, div.container {
    display: none;
  }
  p.messageMobile {
    display: block;
  }
  

}


label.form-label {
  width: 20%;
}

input#greeting_text {
  width: 80%;
  margin-left: 10px;
}

.dropdown .btn-group .dropdown-toggle {
  color: #450063;
  font-weight: 500;
  font-size: 20px;
}


/* width */
div.photos-results::-webkit-scrollbar {
  width: 20px;
}
 
/* Handle */
div.photos-results::-webkit-scrollbar-thumb {
  background: grey; 
  border-radius: 10px;
}

/* Handle on hover */
div.photos-results::-webkit-scrollbar-thumb:hover {
  background: darkgrey; 
}

div.dropdown {
  font-size: 24px;
  /* background: white; */
  padding: 5px;
}

div.display-value {
  display: inline-block;
  flex-wrap: wrap;
  padding: 2.5px;
}

div.display-value p {
  margin: 0;
  font-size: 24px;
  background: white;
  /* padding:5px; */
  border-radius: 10px;
    padding-right: 5px;
    padding-left: 5px;
}

div.dropdown a {
  margin-right: 0;
  border-radius: 10px!important;
}

.image-container {
  display: inline-block;
  position: relative;
}

.image-container svg {
  position: absolute;
  right: 0;
  opacity: .8;
  /* padding: 3px; */
  border: 3px solid white;

}

.image-container ul.menu {
  display: none;
  visibility: hidden;
}

.image-container ul.menu li{
  cursor: pointer;
}

button.save {
  float: right;
  background: #a8a4b9;
  color: white;
}

button.reset {
  background: #301f84;
  color: white;
}

/* button.save:hover, button.reset:hover {
  opacity: 0.5;
} */