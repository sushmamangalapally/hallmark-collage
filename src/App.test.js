import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import SearchForm from './SearchForm';
import '@testing-library/react'
import '@testing-library/jest-dom'
import {
  render,
  screen,
  fireEvent,
  // act,
} from '@testing-library/react';
import axios from 'axios';

jest.mock('axios');

const pictureOne = 
  {
    "id": "ScEXB6uxOVc",
    "created_at": "2020-10-03T16:51:33-04:00",
    "updated_at": "2020-10-14T01:04:40-04:00",
    "promoted_at": null,
    "width": 3382,
    "height": 4793,
    "color": "#171819",
    "blur_hash": "LfNJ]_%2V@WAGdRkNGay$xRkRjkC",
    "description": null,
    "alt_description": null,
    "urls": {
      "raw": "https://images.unsplash.com/photo-1601758282715-6446bbde7082?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjcyNTI1fQ",
      "full": "https://images.unsplash.com/photo-1601758282715-6446bbde7082?ixlib=rb-1.2.1&q=85&fm=jpg&crop=entropy&cs=srgb&ixid=eyJhcHBfaWQiOjcyNTI1fQ",
      "regular": "https://images.unsplash.com/photo-1601758282715-6446bbde7082?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjcyNTI1fQ",
      "small": "https://images.unsplash.com/photo-1601758282715-6446bbde7082?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjcyNTI1fQ",
      "thumb": "https://images.unsplash.com/photo-1601758282715-6446bbde7082?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjcyNTI1fQ"
    },
    "links": {
      "self": "https://api.unsplash.com/photos/ScEXB6uxOVc",
      "html": "https://unsplash.com/photos/ScEXB6uxOVc",
      "download": "https://unsplash.com/photos/ScEXB6uxOVc/download",
      "download_location": "https://api.unsplash.com/photos/ScEXB6uxOVc/download"
    },
    "categories": [],
    "likes": 70,
    "liked_by_user": false,
    "current_user_collections": [],
    "user": {
      "id": "21uOSEd-cSI",
      "updated_at": "2020-10-18T21:20:10-04:00",
      "username": "chewy",
      "name": "Chewy",
      "first_name": "Chewy",
      "last_name": null,
      "twitter_username": "chewy",
      "portfolio_url": "https://www.chewy.com/",
      "bio": "There are endless ways #PetsBringUsTogether. Weâ€™re just here to help.",
      "location": null,
      "links": {
        "self": "https://api.unsplash.com/users/chewy",
        "html": "https://unsplash.com/@chewy",
        "photos": "https://api.unsplash.com/users/chewy/photos",
        "likes": "https://api.unsplash.com/users/chewy/likes",
        "portfolio": "https://api.unsplash.com/users/chewy/portfolio",
        "following": "https://api.unsplash.com/users/chewy/following",
        "followers": "https://api.unsplash.com/users/chewy/followers"
      },
      "profile_image": {
        "small": "https://images.unsplash.com/profile-1600206400067-ef9dc8ec33aaimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        "medium": "https://images.unsplash.com/profile-1600206400067-ef9dc8ec33aaimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        "large": "https://images.unsplash.com/profile-1600206400067-ef9dc8ec33aaimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128"
      },
      "instagram_username": "chewy",
      "total_collections": 0,
      "total_likes": 0,
      "total_photos": 50,
      "accepted_tos": true
    },
  };

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('SearchForm', () => {
  const searchFormProps = {
    searchTerm: 'dog',
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn(),
  };
  const emptySearchFormProps = {
    searchTerm: '',
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn(),
  };

  test('renders the input field with its value', () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByDisplayValue('dog')).toBeInTheDocument();
  });

  test('renders the correct placeholder', () => {
    render(<SearchForm {...emptySearchFormProps} />);

    expect(screen.getByPlaceholderText('Search Pictures')).toHaveValue('')
  });

  test('calls onSearchInput on input field change', () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.change(screen.getByDisplayValue('dog'), {
      target: { value: 'puppies' },
    });

    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
  });

  test('renders snapshot', () => {
    const { container } = render(<SearchForm {...searchFormProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});