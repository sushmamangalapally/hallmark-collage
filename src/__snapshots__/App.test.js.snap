// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders snapshot 1`] = `
<input
  class="form-input"
  id="adding_photos_text"
  placeholder="Search Pictures"
  type="text"
  value="dog"
/>
`;
